agent/engine/task_manager.go

```diff
@@ -765,8 +767,13 @@ func (mtask *managedTask) progressTask() {
 	changed := mtask.UpdateStatus()
 	if changed {
 		seelog.Debugf("Managed task [%s]: container or resource change also resulted in task change", mtask.Arn)
+
 		// If knownStatus changed, let it be known
-		mtask.emitTaskEvent(mtask.Task, "")
+		var taskStateChangeReason string
+		if mtask.GetKnownStatus().Terminal() {
+			taskStateChangeReason = mtask.Task.GetTerminalReason()
```

Comment:
since we emit a STSC on a task status change, we only populate the terminalReason if the the task is at terminal. Currently, we wont bother population the field if it is a transition earlier in the task lifecycle. 

However I can see why we could also avoid this check altogether and just populate the field. We can just use the empty string it the status is a non terminal transition. 

What do you think?